lib/validate_params/param_builder.rb

```diff
@@ -3,27 +3,57 @@
 module ValidateParams
   module Validatable
     class ParamBuilder
-      def initialize(parent_field: nil, validations:)
-        @parent_field = parent_field
-        @validations = validations
-      end
+      Validation = Struct.new(:field, :type, :options, :children, :parent) do
+        def valid?(value, errors)
+          ParamValidator.call(
+            validation: self,
+            value: value,
+            errors: errors
+          )
 
-      def param(field, type, options = {}, &block)
-        if block
-          yield(ParamBuilder.new(parent_field: field, validations: @validations))
-        else
-          @validations << build_config(field, type, options)
+          if children.any?
+            case type.to_s
+            when "Hash"
+              # Skip in case hash is configured and string is passed
+              unless value.is_a?(String)
+                children.each { |c| c.valid?(value&.[](c.field), errors) }
+              end
+            when "Array"
+              values = value ? Array(value) : [nil]
```

Comment:
Maybe it would be better to use `Array.wrap(...)` here?

Because:
```ruby
Array({123 => "asd"}) # => [[123, "asd"]]
Array.wrap({123 => "asd"}) # => [{123=>"asd"}]
```

Reply:
Unfortunately we cannot use this `Array.wrap` method here, since we have a case when the value is `nil` but we need to run validation for it anyway. `Array.wrap(nil) => []` returns an empty list and therefore iteration through it will be impossible.

Reply:
@psyxoz I mean to have a logic same as now just replace `Array(value)`:
```suggestion
              values = value ? Array.wrap(value) : [nil]
```